AcsfToolsUtils.php

```diff
@@ -308,4 +264,117 @@ public function isOperationBlocked(array $data): bool {
     }
     return FALSE;
   }
+
+  /**
+   * Utility function to retrieve a list of sites in an ACSF Factory.
+   *
+   * @return array|bool
+   */
+  public function getAcsfSites(): array|bool {
+    $sites = FALSE;
+
+    // Exit early if the command is executed outside an ACSF server and no
+    // alias is provided.
+    if (isset($this->aliasRecord) && $this->aliasRecord->isLocal() && !$this->checkAcsfFunction('gardens_site_data_load_file')) {
+      return $sites;
+    }
+
+    $map = FALSE;
+    if ($this->checkAcsfFunction('gardens_site_data_load_file')) {
+      $map = gardens_site_data_load_file();
+    }
+    elseif ($this->aliasRecord && !$this->aliasRecord->isLocal() && $sites_json_filepath = $this->getLocalSitesJsonFilepath()) {
+      $alias_name = str_replace('@', '', $this->aliasRecord->name());
+
+      // Download the remote ACSF sites.json so we can process it locally.
+      if (!file_exists($sites_json_filepath)) {
+        $self = $this->siteAliasManager()->getSelf();
+        $process = Drush::drush($self, 'rsync', [$this->aliasRecord->name() . ':/mnt/files/' . $alias_name . '/files-private/sites.json',  $sites_json_filepath, '-y']);
+
+        try {
+          $process->mustRun();
+        }
+        catch (\Exception $e) {
+          return FALSE;
+        }
+      }
+
+      $json = @file_get_contents($sites_json_filepath);
+      $map = $json ? json_decode($json, TRUE) : FALSE;
+    }
+
+    // Look for list of sites and loop over it.
+    if ($map && isset($map['sites'])) {
+      // Acquire sites info.
+      $sites = array();
+      foreach ($map['sites'] as $domain => $site_details) {
+        if (!isset($sites[$site_details['name']])) {
+          $sites[$site_details['name']] = $site_details;
+        }
+
+        // Path domains need a trailing slash to be recognized as a drush alias.
+        if (FALSE !== strpos($domain, '/')) {
+          $domain = rtrim($domain, '/') . '/';
+        }
+
+        $sites[$site_details['name']]['domains'][] = $domain;
+
+        // Identify the site machine name from the acsitefactory.com domain.
+        $machine_name = [];
+        if (preg_match('/(.*)\..*\.acsitefactory\.com/', $domain, $machine_name)) {
+          $sites[$site_details['name']]['machine_name'] = $machine_name[1];
+        }
+      }
+    }
+    else {
+      $this->logger()->error("\nFailed to retrieve the list of sites of the factory.");
+    }
+
+    return $sites;
+  }
+
+  /**
+   * Utility function to retrieve a list of sites in a multi-site Drupal installation.
+   *
+   * @return array
+   * @throws \RuntimeException
+   */
+  public function getMultiSiteSites(): array {
+    // Read information from the sites.php file inside `/sites` directory.
+    // @todo: Change this to proper file where data is available.
+    $multisite_directory_path = DRUPAL_ROOT . '/sites';
+    $multisite_file_path = $multisite_directory_path.'/sites.php';
```

Comment:
Currently reading data from `docroot/sites/sites.php` file. This can be changed if we have different place for reading the information

Reply:
sample entry I have in my sites.php - 
```
$sites['site2.ddev.site'] = 'site2';
$sites['site3.ddev.site'] = 'site3';
```

where site2 and site3 are multisite directories